### PR TITLE
Prevent duplicate files

### LINKED ISSUE #104: Too many duplicate data in zim
zimcheck outputs thousands of duplicate content article from the latest phzh ZIM.

```
[WARNING] Redundant data found:
  course/core-english-01/topic-5-my-feelings-and-myself/unit-1-dealing-with-different-emotions/self-evaluation/05-01-04-intro-evaluation/00-eval-icon.svg (idx 1129) and course/core-english-01/topic-5-my-feelings-and-myself/unit-2-what-makes-me-happy-what-makes-you-sad/self-evaluation/05-02-04-intro-evaluation/00-eval-icon.svg (idx 1170)
  course/core-english-01/topic-5-my-feelings-and-myself/unit-1-dealing-with-different-emotions/self-evaluation/05-01-04-intro-evaluation/00-eval-icon.svg (idx 1129) and course/core-english-01/topic-5-my-feelings-and-myself/unit-3-expressing-my-worries-and-locking-up-my-fears/self-evaluation/05-03-04-intro-evaluation/00-eval-icon.svg (idx 1206)
  course/core-english-01/topic-5-my-feelings-and-myself/unit-1-dealing-with-different-emotions/self-evaluation/05-01-04-intro-evaluation/00-eval-icon.svg (idx 1129) and course/core-english-01/topic-7-how-i-live-together-with-others/unit-1-who-can-help/self-evaluation/07-01-04-intro-evaluation/00-eval-icon.svg (idx 1428)
  course/core-english-01/topic-5-my-feelings-and-myself/unit-1-dealing-with-different-emotions/self-evaluation/05-01-04-intro-evaluation/00-eval-icon.svg (idx 1129) and course/core-english-01/topic-7-how-i-live-together-with-others/unit-2-how-i-can-solve-conflicts/self-evaluation/07-02-04-intro-evaluation/00-eval-icon.svg (idx 1461)
  course/core-english-01/topic-5-my-feelings-and-myself/unit-1-dealing-with-different-emotions/self-evaluation/05-01-04-intro-evaluation/00-eval-icon.svg (idx 1129) and course/core-english-01/topic-7-how-i-live-together-with-others/unit-3-rules-help-us-to-live-together/self-evaluation/07-03-04-intro-evaluation/00-eval-icon.svg (idx 1504)
```

### PR DESCRIPTION
**Pull request by satyamtg**
This fixes #104 by downloading everything to `instance_assets` instead of downloading to different places. 
Also prevents creation of dummy subtitles if content was `{}`, which resulted in many duplicates in `zimcheck`

Please do note that this prevents duplicate files on the basis of unique URLs, however there are some small number of files which have different URLs but same content, and hence are reported by `zimcheck`. There's no way of identifying them without calculating a checksum. I think we shall see https://github.com/openzim/python_scraperlib/issues/33 and use that instead for a small subset of files (which are basically some images under `instance_assets`).

### CODE REVIEW

File: openedx2zim/html_processor.py [hunk 1]
```diff
@@ -149,6 +149,18 @@ def download_images_from_html(self, html_body, output_path, path_from_html):
                         img.attrib["style"] = " max-width:100%"
         return bool(imgs)
 
+    def calculate_root_from_asset(self, path_from_html, root_from_html):
```
Comment (rgaudin) [inline openedx2zim/html_processor.py:152]: I think `get_root_from_asset` is more appropriate here

File: openedx2zim/html_processor.py [hunk 2]
```diff
@@ -149,6 +149,18 @@ def download_images_from_html(self, html_body, output_path, path_from_html):
...
+        """ get path to root from the downloaded/generated asset """
+
+        nb_jumps_root_from_html = root_from_html.count("../")
+        nb_back_jumps_output_path = path_from_html.count("../")
+        path_without_back_jumps = path_from_html[(nb_back_jumps_output_path) * 3 :]
```
Comment (rgaudin) [inline openedx2zim/html_processor.py:157]: Where is this `3` coming from? Maybe add a comment.

File: openedx2zim/html_processor.py (hunk 1 above)
Comment (satyamtg) [reply openedx2zim/html_processor.py]: Yup. Sounds better.

File: openedx2zim/html_processor.py (hunk 2 above)
Comment (satyamtg) [reply openedx2zim/html_processor.py]: That's the length of one back jump, i.e. `len("../")` . Have used that instead, also added a comment.

**Comment by satyamtg** [comment]
> Should we add a function that returns the `"../" * x`? Might bit a bit more readable?

Have added a function `get_back_jumps(nb_jumps)` in utils.py

**Pull request merged**